account_bill_line_distribution/models/account_invoice.py

```diff
@@ -84,23 +84,24 @@ def invoice_validate(self):
                                     'partner_id':
                                         inv.company_id.partner_id.id})
                                 # Debit Account of invoice line
-                                account_obj = self.env['account.account']
-                                code = \
-                                    account_obj.browse(line['account_id']).code
-                                account = account_obj.sudo().search(
-                                    [('code', '=', code),
-                                     ('company_id', '=', company.id)])
+                                comps = self.env['res.company'].sudo().\
+                                    search([('due_from_account_id', '!=', False)])
+                                due_from_accounts = [comp.due_from_account_id.id for comp in comps]
+                                for a_line in from_lines:
+                                    if a_line['account_id'] not in due_from_accounts:
+                                        account_id = a_line['account_id']
```

Comment:
@osi-scampbell I think we can avoid this loop as we can get the code from the account at line 54.

Reply:
@max3903 That is the value at line 52 however, this section of code is outside of that loop so I do not have access to that specific invoice_line_id when we are in that line of code.